### DISCUSSION TITLE
Different sized features per node

### DISCUSSION THREAD

**Discussion by afonso-sousa** [pyg-team/pytorch_geometric#5794] [Q&A]
Hi. I have a graph of nodes with different numbers of features (basically because the features are the tokenization ids of words and may vary in length for subword tokenizers). How can I pad the missing features? I do so for each graph but when collating it will not work. Do I need a custom collate function or is there a way to allow for graphs with features of different sizes?
Before passing the data through the model I will take does features and check their corresponding embeddings, so the passing to the model is not an issue. Just the building of a dataloader. Thanks in advance for any help you can provide.

**Comment by afonso-sousa** [comment]
I have this solution to pad to a `max_len` bigger than any possible case, but was wondering if it could be done with an existing collate function:
```
x[0] = nn.ConstantPad1d((0, max_len - x[0].shape[0]), -1)(x[0])
x = pad_sequence(x, batch_first=True, padding_value=-1)
```

**Reply by rusty1s** [reply]
One thing you could do is override the [`collate_fn`](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/loader/dataloader.py#L10) such that you only pad based on max length of features in your mini-batch. We do not provide automatic support for this, but extending PyG here with your own logic should be straightforward.
